packages/fsm/src/fsm_node.py

```diff
@@ -33,13 +35,20 @@ def __init__(self):
         # Provide service
         self.srv_state = rospy.Service("~set_state", SetFSMState, self.cbSrvSetState)
 
-        # Construct publishers
-        self.pub_dict = dict()
+        # Construct service calls
+        self.srv_dict = dict()
         nodes = rospy.get_param("~nodes")
-
+        # rospy.loginfo(nodes)
         self.active_nodes = None
-        for node_name, topic_name in list(nodes.items()):
-            self.pub_dict[node_name] = rospy.Publisher(topic_name, BoolStamped, queue_size=1, latch=True)
+        
+        # for node_name, topic_name in list(nodes.items()):
+        #     self.pub_dict[node_name] = rospy.Publisher(topic_name, BoolStamped, queue_size=1, latch=True)
+        
+        for node_name, service_name in list(nodes.items()):
+                rospy.wait_for_service(service_name) #  Not sure if there is a better way to do this
```

Comment:
There could be a better way to do this? We probably have to wait for only the first service. But if it tries to subscribe to a service/call it it crashes.